lib/context.js

```diff
@@ -64,6 +65,22 @@ class Context {
     return this.payload.sender.type === 'Bot'
   }
 
+  /**
+   * Returns a github API object or throws an error, if the application could
+   * not be authenticated (for example during the "installation.deleted" event).
+   * @type {github}
+   */
+  get github () {
+    if (!this.api) {
+      throw new Error(
+        'You are trying to make a GitHub request but is seems like your ' +
+        'GitHub App has just been deleted. We cannot authenticate your request.'
+      )
+    }
+
+    return this.api
+  }
+
```

Comment:
I'm wondering if it still makes sense to return an un-authenticated GitHub client here. I can't currently think of a reason why someone would still need `context.github` inside an `installation.deleted` event, but It would be a lot less logic just pass the unauthenticated client rather than special case this.